### PR TITLE
Add get trending platforms route

### PR DESCRIPTION
**Pull request by andreiduque**
## What this PR introduces?

Adds get trending platforms route

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] I followed GitFlow pattern to create the branch
- [ ] My code produces no warnings or errors in the console
- [ ] Tests for the changes have been added / updated
- [ ] Mocks fot the database have been added / updated

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Documentation update
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] CI/CD related changes
```

## Does this PR introduce a breaking change?

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/database/utils/get-trending.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+export const makeGetTrending =
+	({
+		connection,
+		dataTable,
+		idColumn,
+		counterTable,
+		counterIdColumn,
+		counterCounterColumn,
+	}: MakeGetTrending) =>
+	async <T>({ limit, offset }: GetTrendingInput): Promise<Array<T>> => {
```
Comment (henriqueleite42) [inline src/database/utils/get-trending.ts:26]: My bad, this `<T>` should be at the parent function, not this one.
Remove from here.

File: src/database/repositories/platform/get-trending/index.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { makeGetTrending } from "database/utils/get-trending";
+import { MakeRepository } from "database/utils/types";
+
+export const getTrending = ({ connection }: MakeRepository) =>
+	makeGetTrending({
```
Comment (henriqueleite42) [inline src/database/repositories/platform/get-trending/index.ts:5]: `makeGetTrending<GameEntity>({`

File: src/tests/api/platform/get-trending/validate.spec.ts [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+			}
+
+			expect(result instanceof CustomError).toBeTruthy();
+			expect(result.message).toBe(
+				"page must be a `number` type, but the final value was: `NaN` (cast from the value `{}`).",
+			);
+			expect(result.statusCode).toBe(StatusCodeEnum.BAD_REQUEST);
+		});
+	});
+});
```
Comment (henriqueleite42) [inline src/tests/api/platform/get-trending/validate.spec.ts:99]: Add undefined params test

Comment (henriqueleite42) [inline src/tests/api/platform/get-trending/validate.spec.ts:99]: Add test to check what happens if limit exceeds the maximum limit

**Review by henriqueleite42 — CHANGES_REQUESTED** [review]
